src/test/java/com/uid2/shared/auth/AuthorizableStoreTest.java

```diff
@@ -5,74 +5,181 @@
     }
 
     @Test
-    public void getFromKeyHash_returnsClientKey_withValidKeyHash() {
-        String client1KeyHash = this.store.getFromKey("test-key-1").getKeyHash();
+    public void getAuthorizableByKey_returnsNull_withMissingSiteId() {
+        ClientKey client = clientKeyStore.getAuthorizableByKey("UID2-C-L-abcdef.ghijklghijklghijklghijklghijklghijklgh");
 
-        ClientKey client = this.store.getFromKeyHash(client1KeyHash);
-        assertEquals("client1", client.getName());
+        assertNull(client);
+    }
+
+    @Test
+    public void getAuthorizableByKey_returnsNull_withInvalidKeyFormat() {
+        ClientKey client = clientKeyStore.getAuthorizableByKey("invalid-key");
+
+        assertNull(client);
+    }
+
+    @Test
+    public void getAuthorizableByHash_returnsClientKey_withValidKeyHash() {
+        String client1KeyHash = clientKeyStore.getAuthorizableByKey(SITE_11_CLIENT_KEY).getKeyHash();
+        ClientKey client = clientKeyStore.getAuthorizableByHash(client1KeyHash);
+
+        assertEquals("client11", client.getName());
     }
 
     @Test
-    public void getFromKeyHash_returnsNull_withInvalidBase64KeyHash() {
-        ClientKey client = this.store.getFromKeyHash("invalid-key-hash");
+    public void getAuthorizableByHash_returnsNull_withInvalidBase64KeyHash() {
+        ClientKey client = clientKeyStore.getAuthorizableByHash("invalid-key-hash");
+
         assertNull(client);
     }
 
     @Test
-    public void getFromKeyHash_returnsNull_withUnknownKeyHash() {
-        ClientKey client = this.store.getFromKeyHash("invalidkeyhash");
+    public void getAuthorizableByHash_returnsNull_withUnknownKeyHash() {
+        ClientKey client = clientKeyStore.getAuthorizableByHash("unknown");
+
         assertNull(client);
     }
 
     @Test
     public void refresh_returnsNewClients_afterRefresh() {
-        KeyHasher keyHasher = new KeyHasher();
-        KeyHashResult clientHash1 = keyHasher.hashKey("test-key-4");
-        ClientKey client4 = new ClientKey("", clientHash1.getHash(), clientHash1.getSalt(), "", "client4");
+        ClientKey client4 = createClientKey(KEY_HASHER.hashKey(SITE_14_CLIENT_KEY), "client14", 14);
+        clientKeyStore.refresh(List.of(client4));
+
+        assertAll(
+                "refresh returns new clients after refresh",
+                () -> assertEquals("client14", clientKeyStore.getAuthorizableByKey(SITE_14_CLIENT_KEY).getName()),
+                () -> assertNull(clientKeyStore.getAuthorizableByKey(SITE_11_CLIENT_KEY))
+        );
+    }
 
-        this.store.refresh(List.of(client4));
+    private ClientKey createClientKey(KeyHashResult khr, String name, int siteId) {
+        return new ClientKey("", khr.getHash(), khr.getSalt(), "", name, NOW, Set.of(), siteId);
+    }
 
-        assertEquals("client4", this.store.getFromKey("test-key-4").getName());
-        assertNull(this.store.getFromKey("test-key-1"));
+    private static class SitelessAuthorizable implements IAuthorizable {
```

Comment:
Why do we need this? Can't we just have a ClientKey with a legacy key?

Reply:
This was to basically test AdminUser-type authorizables as well, or I guess this can be replaced with a null-site ID client key?

Reply:
Yeah whatever works, I more so meant we should use a plaintext key without a site ID in some test cases. The actual class doesn't matter much

Reply:
Yep, I've added two tests:
1. The one that's essentially the legacy client key (`getAuthorizableByKey_returnsClientKey_withLegacyFormat`)
2. The one that emulates a null-site ID authorizable like AdminUser (`getAuthorizableByKey_returnsKey_withNoSiteId`)